### PR TITLE
Updating ConfigMap to make use of CruResource

### PR DESCRIPTION
**Pull request by codyrancher**
Note:
I don't think there was any useful validation that could be done for configmap. If anyone knows of any please let me know. Right now the only thing that seems to be invalid is an empty name which already has a friendly message when saving.

rancher/dashboard#842

Screen Shot 2020-08-12 at 2 38 16 PM
Screen Shot 2020-08-12 at 2 38 27 PM

### CODE REVIEW

File: edit/configmap.vue [hunk 1]
```diff
@@ -1,70 +1,99 @@
...
           />
         </Tab>
-      </template>
-    </ResourceTabs>
-
-    <Footer :mode="mode" :errors="errors" @save="save" @done="done" />
-  </form>
+        <Tab name="binary-data" :label="t('configmap.tabs.binaryData.label')" :weight="2">
+          <div v-if="!isView || Object.keys(value.data||{}).length" class="row">
+            <KeyValue
```
Comment (vincent99) [inline edit/configmap.vue:69]: Editing non-ascii-like binary data should show you the length and not the raw value.

image

File: edit/configmap.vue [hunk 2]
```diff
@@ -1,70 +1,99 @@
...
+              :read-multiple="true"
+              :value-binary="true"
+              :value-base64="true"
+              :value-can-be-empty="true"
+              :initial-empty-row="true"
+            />
+          </div>
+        </Tab>
+        <Tab
+          name="labels-and-annotations"
```
Comment (codyrancher) [inline edit/configmap.vue:85]: While working on the namespace page I also realized this needs to be rendered conditionally.

v-if="!isView"

File: edit/configmap.vue [hunk 3]
```diff
@@ -1,70 +1,99 @@
...
+          :label="t('generic.labelsAndAnnotations')"
+          :weight="1000"
+        >
+          <Labels
+            default-container-class="labels-and-annotations-container"
+            :spec="value.spec"
```
[the remark this answers is not in the document]
Comment (codyrancher) [reply edit/configmap.vue:91]: I addressed the 'misleading' part here https://github.com/rancher/dashboard/pull/1021 by changing :spec to :value.

I looked through some of the docs about spec. Just for my own information, will a spec field always be accompanied by a status field to let you know if the :spec(desired state) is being met?

[the remark this answers is not in the document]
Comment (vincent99) [reply edit/configmap.vue:91]: They're not strictly tied together AFAIK but generally yes.

Simple direct data resources like configmap/secret just have fields at the top level.

Things that asynchronously manage something (e.g. workloads, things that take time for a process to provision) have a `spec` describing the desired state and `status` and/or `conditions` to tell you what's happening (which we aggregate into `metadata.state`).

Everything has `metadata` with the name, namespace (if the type is namespaced), labels and annotations.

**Comment by westlywright** [comment]
I don't have any additional feedback other than the labels change you're making in rancher/dashboard#1021 will solve an issue I was debugging this morning.

**Comment by vincent99** [comment]
You kind of make it sound like you've fixed things here but I don't see any changes...

File: edit/configmap.vue (hunk 1 above)
Comment (codyrancher) [reply edit/configmap.vue]: Fixed
Screen Shot 2020-08-17 at 12 43 05 PM

**Comment by codyrancher** [comment]
I can see what you mean. My intent was to redirect attention to the other PR and have that merged first. Everything should be fixed here now.

**Pull request merged**